src/app/(proper_react)/(redesign)/(public)/SignUpForm.tsx

```diff
@@ -29,16 +30,35 @@ export const SignUpForm = (props: Props) => {
   const l10n = useL10n();
   const [emailInput, setEmailInput] = useState("");
   const record = useTelemetry();
+  const [cookies] = useCookies(["attributionsFirstTouch"]);
+  const attributionSearchParams = new URLSearchParams(
+    cookies.attributionsFirstTouch,
+  );
+  attributionSearchParams.set(
+    "entrypoint",
+    "monitor.mozilla.org-monitor-product-page",
+  );
+  // This passes an `?email=` query parameter to FxA, causing it to prefill
+  // the email address in the sign-up form. See
+  // https://mozilla.github.io/ecosystem-platform/relying-parties/reference/query-parameters#email
+  attributionSearchParams.set("email", emailInput);
+  attributionSearchParams.set("form_type", "button");
+  if (!attributionSearchParams.has("utm_source")) {
+    attributionSearchParams.append("utm_source", "product");
+  }
+  if (!attributionSearchParams.has("utm_medium")) {
+    attributionSearchParams.append("utm_medium", "monitor");
+  }
+  if (!attributionSearchParams.has("utm_campaign")) {
+    attributionSearchParams.append("utm_campaign", "get_free_scan");
+  }
 
   const onSubmit: FormEventHandler = (event) => {
     event.preventDefault();
     void signIn(
       "fxa",
       { callbackUrl: props.signUpCallbackUrl },
-      // This passes an `?email=` query parameter to FxA, causing it to prefill
-      // the email address in the sign-up form. See
-      // https://mozilla.github.io/ecosystem-platform/relying-parties/reference/query-parameters#email
-      { email: emailInput },
+      attributionSearchParams.toString(),
```

Comment:
Looking at the type definition of `signIn`, it looks like it also accepts the `URLSearchParams` object directly? It's probably converting it back to it internall anyway.

```suggestion
      attributionSearchParams,
```

Reply:
Tried without `toString()` and it actually doesn't work. Reverting it back